# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1337311173 14400
#      Thu May 17 23:19:33 2012 -0400
# Node ID 7c463e49006209079f9ca83bb6f5d0169ff9c089
# Parent  f40c355491cc0e24b25cd65bec03bc172567c7c6
* lex.ll: fix paren mismatch problem in previous change.

diff --git a/src/lex.ll b/src/lex.ll
--- a/src/lex.ll
+++ b/src/lex.ll
@@ -1524,17 +1524,17 @@ is_keyword_token (const std::string& s)
         case until_kw:
           break;
 
         case end_kw:
           if (inside_any_object_index ()
               || (! reading_classdef_file
                   && (lexer_flags.defining_func
                       && ! (lexer_flags.looking_at_return_list
-                            || lexer_flags.parsed_function_name.top ())))
+                            || lexer_flags.parsed_function_name.top ()))))
             return 0;
 
           yylval.tok_val = new token (token::simple_end, l, c);
           lexer_flags.at_beginning_of_statement = true;
           break;
 
         case end_try_catch_kw:
           yylval.tok_val = new token (token::try_catch_end, l, c);
